### PR TITLE
[web] Use webargs instead of flask_restful.reqparse

### PR DESCRIPTION
**Pull request by FestplattenSchnitzel**

### CODE REVIEW

**Comment by lukasjuhrich** [comment]
Warning: I've squashed your `fixup!` commits and force-pushed.

File: pycroft/lib/swdd.py [hunk 1]
```diff
@@ -13,7 +14,7 @@
 swdd_hmac_key = os.environ.get('SWDD_HASH_KEY')
 
 
-def get_swdd_person_id(first_name: str, last_name: str, birthdate: str) -> int | None:
```
Comment (lukasjuhrich) [inline pycroft/lib/swdd.py:16]: Please do API changes (such as the datatype of a passed argument into this lib function) in a separate commit in the future to make verification / review easier. Fine for now, I'll just have to check all the usages.

Comment (lukasjuhrich) [reply pycroft/lib/swdd.py:16]: Checked everything, looks good.

File: web/api/v0/__init__.py [hunk 2]
```diff
@@ -343,19 +389,19 @@ def post(self, user_id: int) -> ResponseReturnValue:
...
             abort(412, message="User already has a host with interface.")
 
-        user.birthdate = args.birthdate
+        user.birthdate = birthdate
 
         host = Host.q.filter_by(owner_id=user.id).one_or_none()
 
         try:
-            host_name = args.host_name if args.host_name else None
+            host_name = host_name if host_name else None
```
Comment (lukasjuhrich) [inline web/api/v0/__init__.py:397]: this is a NOP:
```suggestion
```

File: web/api/v0/__init__.py [hunk 3]
```diff
@@ -252,9 +271,13 @@ def post(self) -> ResponseReturnValue:
 
 
 class UserByIPResource(Resource):
-    def get(self) -> ResponseReturnValue:
-        ipv4 = request.args.get('ip', IPAddress)
-
+    @use_kwargs(
+        {
+            "ipv4": fields.IP(required=True, data_key="ip"),
```
Comment (lukasjuhrich) [inline web/api/v0/__init__.py:276]: Not sure what to do about the type warning here; mypy considers the `IP.__init__` function untyped due to lack of `-> None`. nothing to be done about it. I don't want to suppress this error on every invocation tho…
Plus, stubs (`.pyi`) cannot augment existing lib type hints, they act as replacements.

**Comment by lukasjuhrich** [comment]
Looks good, thanks! Find a way to solve the mypy errors (e.g. via localized suppressions) and then do a `merge --no-ff` from `develop`. I want this merged quickly.

**Comment by lukasjuhrich** [comment]
> Find a way to solve the mypy errors (e.g. via localized suppressions) and then do a `merge --no-ff` from `develop`

Did the suppression, please do the merge after CI passes @FestplattenSchnitzel

**Comment by lukasjuhrich** [comment]
No idea why the ldap connection could not be established; the finance test just has some end-of-month flakiness. Rerunning and hoping for the best.

**Comment by lukasjuhrich** [comment]
Sorry for all the push spam (and force-pushes), debugging a non-reproducible error in CI is always a pain.

**Comment by FestplattenSchnitzel** [comment]
No problem; thanks for debugging this.

**Comment by lukasjuhrich** [comment]
Ah, finally. Seemed to be the update of the `openldap` image to `bookworm`.
Something changed with the ppolicy and I couldn't make `dinkel`s base image work in half an hour, so let's just pin it to `<2.5`. Let's hope for running tests.

CI issues are the worst.

**Comment by lukasjuhrich** [comment]
Ah, finally

**Comment by FestplattenSchnitzel** [comment]
This accidentally introduced https://github.com/agdsn/sipa/issues/472 , fixed with https://github.com/agdsn/pycroft/pull/667 .

**Pull request merged**